### ISSUE TITLE
[FEA] Allow pytests to be run if have only binary install (used to work for 0.14)

### ISSUE THREAD

**Issue by pseudotensor** [rapidsai/cusignal#268]
https://github.com/rapidsai/cusignal/issues/217#issuecomment-691670359

Good to be able to validate install via conda, to ensure actually works.  From this comment, at least from 0.15 and above, the pytest seems not possible.  I hit various errors like those mentioned in the issue above (even though they built from source).  However, for 0.14 there is no such problem and all tests can be run.

Any production use of rapids is best if one can validate all tests suites pass before using it for ones own purposes.  Thanks!

**Comment by awthomp** [comment]
Hi @pseudotensor -- thanks for filing an issue on cuSignal! While troubleshooting your bug from a fresh conda env and install, I noticed that, while pytest-benchmark is disabled if you run `pytest` after cloning the library and installing via conda, you'll get errors as it looks like there's a firm dependency on pytest-benchmark, even if it's not used.

Is this what you're seeing? If you run `conda install pytest-benchmark` and then try to run the correctness checks, will the tests pass?

Could you be a bit more specific about the errors you're seeing?

CC @mnicely

**Comment by pseudotensor** [comment]
I have pytest-benchmark installed.  Here's my conda list and errors:

[list.txt](https://github.com/rapidsai/cusignal/files/5368426/list.txt)


```
229 failed, 783 passed, 100 warnings in 81.88s (0:01:21) 


FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestUpFirDn::test_upfirdn_gpu[0-9-3-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestUpFirDn::test_upfirdn_gpu[0-9-7-1-16384] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestUpFirDn::test_upfirdn_gpu[0-9-7-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[64-2048-4096-float32] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[64-2048-4096-float64] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[64-2048-4096-complex64] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[64-2048-4096-complex128] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[128-2048-4096-float32] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[128-2048-4096-float64] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[128-2048-4096-complex64] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[128-2048-4096-complex128] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[256-2048-4096-float32] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[256-2048-4096-float64] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[256-2048-4096-complex64] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_filtering.py::TestFilter::TestChannelizePoly::test_channelizepoly_gpu[256-2048-4096-complex128] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[clip-1--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[clip-1--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[clip-2--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[clip-2--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[wrap-1--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[wrap-1--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[wrap-2--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmin::test_argrelmin_gpu[wrap-2--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[clip-1--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[clip-1--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[clip-2--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[clip-2--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[wrap-1--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[wrap-1--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[wrap-2--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelmax::test_argrelmax_gpu[wrap-2--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[clip-1--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[clip-1--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[clip-2--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[clip-2--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[wrap-1--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[wrap-1--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[wrap-2--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[wrap-2--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-True-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-True-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-False-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-False-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-True-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-True-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-False-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-False-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
```

Just sample of the failures, the errors are all the same.

FYI cudf, rmm, dask-cuda, cuml, etc. all pass (apart from 1 cuml failure).

**Comment by mnicely** [comment]
Hi @pseudotensor, thanks for taking the time to post the information above. I've just tested 0.14, 0.15, and nightly binaries in a fresh conda environment with not issues. Do mind providing how you installed cuSignal and how you ran pytest?

**Comment by pseudotensor** [comment]
https://github.com/rapidsai/cuml/issues/2960#issue-719665547

I installed it as shown there.

```
pytest -s -v cusignal/python/cusignal/test
```

**Comment by mnicely** [comment]
It seems you are running pytest against a local directory. With cuSignal 0.15+ binaries, fatbins are loaded where cusignal is installed.

You'll need to run pytest from the installed directory.
```bash
(test) mnicely@mnicely-Blade:~/anaconda3/envs/test/lib/python3.7/site-packages/cusignal$ ll
total 84
drwxrwxr-x 19 mnicely mnicely 4096 Oct 12 21:42 ./
drwxrwxr-x 54 mnicely mnicely 4096 Oct 12 21:39 ../
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 acoustics/
drwxrwxr-x  2 mnicely mnicely 4096 Oct 12 21:42 .benchmarks/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 bsplines/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 convolution/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 estimation/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 filter_design/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 filtering/
-rw-rw-r--  2 mnicely mnicely 2598 Oct 12 03:17 __init__.py
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 io/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 peak_finding/
drwxrwxr-x  2 mnicely mnicely 4096 Oct 12 21:38 __pycache__/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:42 .pytest_cache/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 spectral_analysis/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 test/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 utils/
-rw-rw-r--  2 mnicely mnicely  511 Oct 12 03:22 _version.py
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 waveforms/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 wavelets/
drwxrwxr-x  3 mnicely mnicely 4096 Oct 12 21:38 windows/
(test) mnicely@mnicely-Blade:~/anaconda3/envs/test/lib/python3.7/site-packages/cusignal$ pytest -v --benchmark-disable
=================================================================================== test session starts ===================================================================================
platform linux -- Python 3.7.8, pytest-6.1.1, py-1.9.0, pluggy-0.13.1 -- /home/mnicely/anaconda3/envs/test/bin/python
cachedir: .pytest_cache
benchmark: 3.2.3 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/mnicely/anaconda3/envs/test/lib/python3.7/site-packages/cusignal
plugins: benchmark-3.2.3
collected 1012 items                                                                                                                                                                      

test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_cpu[123-256] PASSED                                                                               [  0%]
test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_cpu[123-16384] PASSED                                                                             [  0%]
test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_cpu[256-256] PASSED                                                                               [  0%]
test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_cpu[256-16384] PASSED                                                                             [  0%]
test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_gpu[123-256] PASSED                                                                               [  0%]
test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_gpu[123-16384] PASSED                                                                             [  0%]
test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_gpu[256-256] PASSED                                                                               [  0%]
test/test_acoustics.py::TestAcoustics::TestComplexCepstrum::test_complex_cepstrum_gpu[256-16384] PASSED                                                                             [  0%]
test/test_acoustics.py::TestAcoustics::TestRealCepstrum::test_real_cepstrum_cpu[123-256] PASSED                                                                                     [  0%]
```

**Comment by pseudotensor** [comment]
Thanks, tried that and still happens:

```
(base) ~/miniconda3/lib/python3.8/site-packages/cusignal$ pytest -s -v ~/cusignal/python/cusignal/test
```
```
FAILED ../../../../../cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[wrap-1--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[wrap-2--1-1-32768] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_peak_finding.py::TestPeakFinding::TestArgrelextrema::test_argrelextrema_gpu[wrap-2--1-2-256] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-True-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-True-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-False-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[True-False-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-True-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-True-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-False-65536-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
FAILED ../../../../../cusignal/python/cusignal/test/test_spectral_analysis.py::TestSpectral::TestLombScargle::test_lombscargle_gpu[False-False-262144-1024] - cupy.cuda.driver.CUDADriverError: CUDA_ERROR_FILE_NOT_FOUND: file not found
============================================================================================== 229 failed, 783 passed, 100 warnings in 328.68s (0:05:28) ==============================================================================================
(base) jon@mr-dl10:~/minicondadai3/lib/python3.8/site-packages/cusignal$
```

**Comment by pseudotensor** [comment]
Ah, but you also specified no test path, I see you have test in the site-packages/cusignal.

Ok, now that works.  Thanks!

`794 passed, 88 warnings in 286.41s (0:04:46)`

**Comment by mnicely** [comment]
I’m confused. In your other issue, it shows you install cuSignal=0.15. 

But the error above show the peak finding functions looking for CUDA, which were just released in 0.16. What version are you testing. 

Can you see if fatbins are in the module directories??

**Comment by pseudotensor** [comment]
Ah, you are right, the cusignal branch for the repo was branch-0.16.

It works if I switch the repo to branch-0.15, so I guess things do work even with the test directory. I must have missed/skipped changing the branch.  Sorry for that!

I think I saw the issue online and just assumed it was related to the comment I reference.

```
(base) ~/miniconda3/lib/python3.8/site-packages/cusignal$ python
Python 3.8.3 (default, May 19 2020, 18:47:26) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cusignal
>>> cusignal.__version__
'0.15.0'
>>> 
```
